src/rp2_common/pico_cyw43_arch/cyw43_arch.c

```diff
@@ -94,7 +100,15 @@ int cyw43_arch_wifi_connect_until(const char *ssid, const char *pw, uint32_t aut
         cyw43_arch_poll();
         cyw43_arch_wait_for_work_until(until);
     }
-    return status == CYW43_LINK_UP ? 0 : status;
+    // Turn status into a pico_error_codes
+    assert(status == CYW43_LINK_UP || (status < 0 && status != CYW43_LINK_NONET));
```

Comment:
is this truly an assert, i.e. no such statuses can happen?

Reply:
It's asserting my understanding of the code. You should only see success (link up) or errors, and you should no longer see the nonet error (it now times out instead). I put it in because the there are some positive status values and a load of negative errors which we're changing to pico_error_codes. I guess I could make the assert even stricter and check we only get the specific errors we're expecting.